src/ripple/overlay/Message.h

```diff
@@ -45,17 +46,35 @@ namespace ripple {
 // MessageType should be a Message class generated by the protobuf compiler.
 //
 
+using namespace compression;
+
 class Message : public std::enable_shared_from_this <Message>
 {
 public:
+    /** Constructor
+     * @param message Protocol message to serialize
+     * @param type Protocol message type
+     */
     Message (::google::protobuf::Message const& message, int type);
 
 public:
-    /** Retrieve the packed message data. */
+    /** Retrieve the packed message data.
+     * @param compressed Request compressed (Compress::On) or uncompressed (Compress::Off) payload buffer
+     * @return Payload buffer
+     */
     std::vector <uint8_t> const&
-    getBuffer () const
+    getBuffer (Compressed compressed)
     {
-        return mBuffer;
+        if (compressed == Compressed::Off)
+            return mBuffer;
+
+        if (!mCompressedRequested)
```

Comment:
Minor nit, but given the usage, this is probably better named: `alreadyCompressed_`.

Also I'm concerned about thread-safety here; I'm not sure it's a problem, but we need to triple-check that it's not. I'd almost feel better if we just did the compression during construction of the message, so that it never became an issue.

Reply:
Changed to `alreadyRequested_`. The message may or may not be compressed.